prime-router/src/main/kotlin/history/azure/DeliveryFacade.kt

```diff
@@ -95,11 +97,22 @@ class DeliveryFacade(
     fun findDetailedDeliveryHistory(
         deliveryId: Long,
     ): DeliveryHistory? {
-        return dbDeliveryAccess.fetchAction(
+        val deliveryHistory = dbDeliveryAccess.fetchAction(
             deliveryId,
             orgName = null,
             DeliveryHistory::class.java
         )
+        val reportId = deliveryHistory?.reportId
+        if (reportId != null) {
+            val roots = reportService.getRootReports(UUID.fromString(reportId))
+            deliveryHistory.originalIngestion = roots.map {
+                mapOf(
+                    "reportId" to it.reportId,
+                    "ingestionTime" to it.createdAt,
```

Comment:
While perhaps not immediately required, I think the following would also be useful for the user:

`"sendingOrg" to it.sendingOrg`

Reply:
Done, @arnejduranovic!